api/v1beta1/cluster_types.go

```diff
@@ -184,6 +194,25 @@ type MachineDeploymentTopology struct {
 	Variables *MachineDeploymentVariables `json:"variables,omitempty"`
 }
 
+// MachineHealthCheckTopology defines a MachineHealthCheck for a group of machines.
+type MachineHealthCheckTopology struct {
+	// Enable controls if a MachineHealthCheck should be created for the target machines.
+	// If false:
+	//		No MachineHealthCheck will be created.
+	// If not set:
+	//		A MachineHealthCheck will be created if it is defined here or in the associated ClusterClass.
+	//		If no MachineHealthCheck is defined then none will be created.
+	// If true:
+	//		A MachineHealthCheck is guaranteed to be created. Cluster validation will block if `enable` is
+	//		true and no MachineHealthCheck definition is available.
```

Comment:
I'm not 100% sure indentation/spacing works in godoc... can you kindly confirm it renders properly?

Reply:
The indentation seems to be working in godoc html that gets generated properly but it is not rendering well in an IDE hover tip. I dropped the indentation and added empty lines to separate out the multiple section. (Empty line is recommended to separate out sections)